logstash-core/lib/logstash/agent.rb

```diff
@@ -106,6 +106,7 @@ def stop_background_services
   def configure_metric
     if collect_metric?
       @logger.debug("Agent: Configuring metric collection")
+      LogStash::Instrument::Collector.instance.agent = self
```

Comment:
We forward this change for later, we will add issue and comment to properly refactor the runner to pass all agent informations.